src/business_logic/execution/execution_entry_point.rs

```diff
@@ -0,0 +1,283 @@
+use std::collections::VecDeque;
+
+use cairo_rs::{
+    types::relocatable::{MaybeRelocatable, Relocatable},
+    vm::{
+        self,
+        runners::cairo_runner::{CairoArg, CairoRunner, ExecutionResources},
+        vm_core::VirtualMachine,
+    },
+};
+use felt::Felt;
+use num_traits::{ToPrimitive, Zero};
+
+use super::{
+    execution_errors::ExecutionError,
+    objects::{CallInfo, CallType, TransactionExecutionContext},
+};
+use crate::{
+    business_logic::{
+        fact_state::in_memory_state_reader::InMemoryStateReader,
+        state::{cached_state::CachedState, state_api_objects::BlockInfo},
+    },
+    services::api::contract_class::{ContractClass, ContractEntryPoint},
+    utils::get_integer_range,
+};
+use crate::{
+    business_logic::{
+        fact_state::state::ExecutionResourcesManager,
+        state::state_api::{State, StateReader},
+    },
+    core::syscalls::{
+        business_logic_syscall_handler::BusinessLogicSyscallHandler,
+        syscall_handler::{self, SyscallHandler, SyscallHintProcessor},
+    },
+    definitions::{
+        constants::TRANSACTION_VERSION,
+        general_config::{self, StarknetGeneralConfig},
+    },
+    services::api::contract_class::EntryPointType,
+    starknet_runner::runner::StarknetRunner,
+    utils::{get_deployed_address_class_hash_at_address, validate_contract_deployed, Address},
+};
+
+/// Represents a Cairo entry point execution of a StarkNet contract.
+#[derive(Debug)]
+pub(crate) struct ExecutionEntryPoint {
+    call_type: CallType,
+    contract_address: Address,
+    code_address: Option<Address>,
+    class_hash: Option<[u8; 32]>,
+    calldata: Vec<Felt>,
+    caller_address: Address,
+    entry_point_selector: Felt,
+    entry_point_type: EntryPointType,
+}
+
+impl ExecutionEntryPoint {
+    pub fn new(
+        contract_address: Address,
+        calldata: Vec<Felt>,
+        entry_point_selector: Felt,
+        caller_address: Address,
+        entry_point_type: EntryPointType,
+        call_type: Option<CallType>,
+        class_hash: Option<[u8; 32]>,
+    ) -> Self {
+        ExecutionEntryPoint {
+            call_type: call_type.unwrap_or(CallType::Call),
+            contract_address,
+            code_address: None,
+            class_hash,
+            calldata,
+            caller_address,
+            entry_point_selector,
+            entry_point_type,
+        }
+    }
+
+    /// Executes the selected entry point with the given calldata in the specified contract.
+    /// The information collected from this run (number of steps required, modifications to the
+    /// contract storage, etc.) is saved on the resources manager.
+    /// Returns a CallInfo object that represents the execution.
+    pub fn execute(
```

Comment:
Can we use references in the function inputs?
like `general_config: &StarknetGeneralConfig`